scout/adapter/mongo/variant.py

```diff
@@ -204,6 +204,19 @@ def variants(
 
         return result
 
+    def variants_count(self, case_id, query=None, variant_ids=None, category="snv"):
+        """Count number of variants
+        Replace deprecated pymongo.cursor.count()"""
+        mongo_query = self.build_query(
+            case_id, query=query, variant_ids=variant_ids, category=category
+        )
+        result = self.variant_collection.count_documents(mongo_query)
+
+        return result
+
+    def get_query(self, case_id, query, variant_ids, category):
```

Comment:
`get_query` was written at a time I thought a universal solution for replacing `cursor.count()` was possible…

Reply:
The design choice was made to be semantic and ideologically consistent with how the database interface is written. Direkt results are never returned from the API. Instead you get "handels" to be used for further actions. Usually a cursor (disguised by the naming convention "xxx_query" leading my thoughts to 1. a list of results or 2. a query dict used for further interactions). 

I'll rewrite it as:
```
    def count_query(self, case_id, query, variant_ids, category):
        """Returns number of variants

        Arguments:
            case_id(str): A string that represents the case
            query(dict): A dictionary with querys for the database
            variant_ids(List[str])
            category(str): 'sv', 'str', 'snv', 'cancer' or 'cancer_sv'

        Returns:
             integer
        """

        query = self.build_query(case_id, query=query, variant_ids=variant_ids, category=category)
        self.variant_collection.count_documents(query)

Reply:
Perhaps move it to the scout/adapter/mongo/query.py file?

Reply:
> Perhaps move it to the scout/adapter/mongo/query.py file?

I don't think that is possible. As `self.variant_collection` need to be referred.

Reply:
I'm quite sure it will work, since variant collection is defined in the setup of the adapter class: https://github.com/Clinical-Genomics/scout/blob/c97a4b1f977fd50e7e4dd92bdf9228beeeff526d/scout/adapter/mongo/base.py#L101

Try to invoke it in the same way (self. variant_collection)

Reply:
Or perhaps import the other class, dunno

Reply:
I don't agree that moving the functionality to 'query.py' is desired. 
1. 'query.py' purpose seems to be acting as a library module, functions are called to return dicts later used for querying.
2. 'query.py' is a mess! Adding more lines of code will only make the much needed refactoring harder.
3. the purpose is counting number of variants -so I'll rename it `count_variants()`

Reply:
Ok!